clang/test/dpct/cusparse.cu

```diff
@@ -471,3 +471,177 @@ void foo7_1(int c_nnz,
+  // CHECK-NEXT: */
+  cusparseDestroyCsrgemm2Info(info_s);
+  cusparseDestroyCsrgemm2Info(info_d);
+  cusparseDestroyCsrgemm2Info(info_c);
+  cusparseDestroyCsrgemm2Info(info_z);
+}
+
+void foo9_2(cusparseMatDescr_t descrB,
+            cusparseMatDescr_t descrC,
+            cusparseMatDescr_t descrD,
+            const float *alpha_s,
+            const float *beta_s,
+            const float* val_a_s,
+            const int* row_ptr_a,
+            const int* col_ind_a,
+            const float* val_b_s,
+            const int* row_ptr_b,
+            const int* col_ind_b,
+            const float* val_d_s,
+            const int* row_ptr_d,
+            const int* col_ind_d,
+            float* val_c_s,
+            int* row_ptr_c,
+            int* col_ind_c,
+            int nnzA,
+            int nnzB,
+            int nnzD,
+            int nnzC,
+            csrgemm2Info_t info_s,
+            void *ws) {
+  // CHECK: dpct::sparse::csrgemm2(handle, 3, 4, 3, alpha_s, descrA, val_a_s, row_ptr_a, col_ind_a, descrB, val_b_s, row_ptr_b, col_ind_b, beta_s, descrD, val_d_s, row_ptr_d, col_ind_d, descrC, val_c_s, row_ptr_c, col_ind_c);
+  cusparseScsrgemm2(handle, 3, 4, 3, alpha_s, descrA, nnzA, val_a_s, row_ptr_a, col_ind_a, descrB, nnzB, val_b_s, row_ptr_b, col_ind_b, beta_s, descrD, nnzD, val_d_s, row_ptr_d, col_ind_d, descrC, val_c_s, row_ptr_c, col_ind_c, info_s, ws);
+}
+
+void foo9_1(cusparseMatDescr_t descrB,
+            cusparseMatDescr_t descrC,
+            cusparseMatDescr_t descrD,
+            const float *alpha_s,
+            const float *beta_s,
+            const float* val_a_s,
+            const int* row_ptr_a,
+            const int* col_ind_a,
+            const float* val_b_s,
+            const int* row_ptr_b,
+            const int* col_ind_b,
+            const float* val_d_s,
+            const int* row_ptr_d,
+            const int* col_ind_d,
+            float* val_c_s,
+            int* row_ptr_c,
+            int* col_ind_c,
+            int nnzA,
+            int nnzB,
+            int nnzD,
+            int nnzC,
+            csrgemm2Info_t info_s,
+            size_t ws_s) {
+#ifndef NO_BUILD_TEST
+  // CHECK: /*
+  // CHECK-NEXT: DPCT1134:{{[0-9]+}}: The tool cannot deduce the consumer API ("dpct::sparse::csrgemm") of this API, and this API has 2 arguments depending on the 8th and the 12th parameters of the consumer API. Please replace the 2 arguments tagged as "dpct_placeholder" with the corresponding value.
+  // CHECK-NEXT: */
+  // CHECK-NEXT: dpct::sparse::csrgemm2_nnz(handle, 3, 4, 3, descrA, nnzA, dpct_placeholder, row_ptr_a, col_ind_a, descrB, nnzB, dpct_placeholder, row_ptr_b, col_ind_b, descrD, nnzD, dpct_placeholder, row_ptr_d, col_ind_d, descrC, row_ptr_c, &nnzC);
+  cusparseXcsrgemm2Nnz(handle, 3, 4, 3, descrA, nnzA, row_ptr_a, col_ind_a, descrB, nnzB, row_ptr_b, col_ind_b, descrD, nnzD, row_ptr_d, col_ind_d, descrC, row_ptr_c, &nnzC, info_s, &ws_s);
```

Comment:
Why is there no call to `cusparseXcsrgemm2_bufferSizeExt` API?

Reply:
Test updated